processor/stream/stream_processor.go

```diff
@@ -46,6 +46,34 @@ type StreamReader interface {
 	LatestLine() []byte
 }
 
+// srErrorWrapper wraps stream decoders and converts errors to
+// something we know how to deal with
+type srErrorWrapper struct {
+	StreamReader
+}
+
+func (s *srErrorWrapper) Read() (map[string]interface{}, error) {
+	v, err := s.StreamReader.Read()
+	if err != nil {
+		if _, ok := err.(decoder.JSONDecodeError); ok {
+			return nil, &Error{
+				Type:     InvalidInputErrType,
+				Message:  err.Error(),
+				Document: string(s.StreamReader.LatestLine()),
+			}
+		}
+
+		if err == decoder.ErrLineTooLong {
+			return nil, &Error{
+				Type:     InvalidInputErrType,
+				Message:  "event exceeded the permitted size.",
```

Comment:
You could also simply use `err.Error` as the message is set in the error, and combine the two error checks as the code is the same.

Reply:
i had that before, but changed it. The Line reader returns an error that says

> Line exceeded permitted length

which is fine for a `LineReader`, but i wanted to give our users a better error message:

> event exceeded the permitted size